tests/tools/onert_train/src/rawdataloader.cc

```diff
@@ -45,33 +79,24 @@ Generator RawDataLoader::loadData(const std::string &input_file, const std::stri
     start += (bufsize_for(&expected_infos[i]) / batch_size * data_length);
   }
 
-  try
-  {
-    _input_file = std::ifstream(input_file, std::ios::ate | std::ios::binary);
-    _expected_file = std::ifstream(expected_file, std::ios::ate | std::ios::binary);
-  }
-  catch (const std::exception &e)
-  {
-    std::cerr << e.what() << std::endl;
-    std::exit(-1);
-  }
-
-  return [input_origins, expected_origins, &input_infos, &expected_infos,
-          this](uint32_t idx, std::vector<Allocation> &inputs, std::vector<Allocation> &expecteds) {
-    for (uint32_t i = 0; i < input_infos.size(); ++i)
-    {
-      auto bufsz = bufsize_for(&input_infos[i]);
-      _input_file.seekg(input_origins[i] + idx * bufsz, std::ios::beg);
-      _input_file.read(reinterpret_cast<char *>(inputs[i].data()), bufsz);
-    }
-    for (uint32_t i = 0; i < expected_infos.size(); ++i)
-    {
-      auto bufsz = bufsize_for(&expected_infos[i]);
-      _expected_file.seekg(expected_origins[i] + idx * bufsz, std::ios::beg);
-      _expected_file.read(reinterpret_cast<char *>(expecteds[i].data()), bufsz);
-    }
-    return true;
-  };
+  return std::make_tuple(
+    [input_origins, expected_origins, &input_infos, &expected_infos,
+     this](uint32_t idx, std::vector<Allocation> &inputs, std::vector<Allocation> &expecteds) {
+      for (uint32_t i = 0; i < input_infos.size(); ++i)
+      {
+        auto bufsz = bufsize_for(&input_infos[i]);
+        _input_file.seekg(input_origins[i] + idx * bufsz, std::ios::beg);
+        _input_file.read(reinterpret_cast<char *>(inputs[i].data()), bufsz);
+      }
+      for (uint32_t i = 0; i < expected_infos.size(); ++i)
+      {
+        auto bufsz = bufsize_for(&expected_infos[i]);
+        _expected_file.seekg(expected_origins[i] + idx * bufsz, std::ios::beg);
+        _expected_file.read(reinterpret_cast<char *>(expecteds[i].data()), bufsz);
+      }
+      return true;
+    },
+    data_length);
```

Comment:
(Optional) This function is too long and has multiple roles. It would be better to have a separate function to get data length.

Reply:
I considered several designs for this. But I couldn't find a better way. I will update this code to increase readability next time. Thank you for feedback. :)